CONTRIBUTING.md

```diff
@@ -55,25 +130,95 @@ Go to <http://localhost:8080> for the example playground.
+  README.md           << the introduction of the project
+```
+
+### The source folder
+
+```
+▸ src/
+  ▸ components/       << The pReact components for the UI of our widgets (UI)
+  ▸ connectors/       << The source of all the connectors (UX without UI)
+  ▸ css/              << The source of the themes (reset / algolia theme)
+  ▸ decorators/       << Preact HoC that factorize some behaviour of the widgets
+  ▸ lib/              << The core of the library, instantsearch, url
+  ▸ shams/            << Object.freeze sham that avoid breaking your project
+  ▸ shims/            << Shim for getPrototypeOf
+  ▸ widgets/          << The source of the widgets (UX + UI)
+```
+
+### The documentation source
 
-We have unit tests written with [Jest](https://facebook.github.io/jest/):
+```
+▾ docgen/
+  ▸ assets/                         << static assets
+  ▸ layouts/                        << pug files that define the structure of the pages
+  ▸ plugins/                        << custom metalsmith plugins
+  ▸ src/                            << Source for the editorial content
+    build.js                        << build script (prod)
+    builder.js                      << main js file of the doc builder
+    config.js                       << variables definiton for the build
+    devServer.js                    << config for the dev server
+    mdRenderer.js                   << customized md renderer
+    middlewares.js                  << metalsmit plugins definitions
+    package.json
+    path.js                         << path config file
+    postcss.config.js               << postcss config file
+    start.js                        << build script (dev)
+    syntaxHighlighting.js           << customized syntax highlighting
+    webpack.config.babel.js         << common webpack config
+    webpack.config.build.babel.js   << prod webpack config
+    webpack.config.start.babel.js   << dev webpack config
+```
+
+### The dev app
+
+```
+▾ dev/
+  ▾ app/                      << the source of the dev app
+    ▸ custom-widgets/         << the source of the custom widgets to validate the connector API
+    ▸ templates/              << templates used for the common widgets
+      index.js                << main script of the dev app
+      init-builtin-widgets.js << initialization of the IS.js app with built-in widgets
+      init-jquery-widgets.js  << initialization of the IS.js app with custom widget / jQuery
+      init-vanilla-widgets.js << initialization of the IS.js app with other custom widgets
+      wrap-with-hits.js       << utility source that adds a hits and searchbox for each story
+    style.css
+    template.html
+    webpack.config.js
+```
 
-Single run and linting:
+## Tests 🔬
+
+### Unit tests
+Our unit tests are written with [Jest](https://facebook.github.io/jest/):
+
+To run a the tests once:
 ```sh
 yarn test
 ```
 
-Watch mode:
+To run the test continuously (useful in dev / debugging):
```

Comment:
To run the tests continuously based on what you changed (useful when developing or debugging):